pkg/controller/namespacelabel/namespacelabel_controller.go

```diff
@@ -133,9 +160,72 @@ func (r *ReconcileUninstall) Reconcile(request reconcile.Request) (reconcile.Res
 				}
 			}
 		}
-		logrus.Info("Reconciling Uninstall completed")
-	}
 
+		if label == "cidr" {
+			logrus.Infof("Cidr value : %v, passed in as a namespace label", value)
+			cfgMap := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      configMapName,
+					Namespace: namespace,
+				},
+			}
+
+			err := r.client.Get(context.TODO(), k8sclient.ObjectKey{Name: configMapName, Namespace: request.NamespacedName.Name}, cfgMap)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			data := []byte(cfgMap.Data["_network"])
+
+			var cfgMapData network
+
+			// Unmarshal or Decode the JSON to the interface.
+			err = json.Unmarshal([]byte(data), &cfgMapData)
+			if err != nil {
+				logrus.Error(err)
+			}
+
+			cidr := cfgMapData.Production.CreateStrategy.CidrBlock
+
+			if cidr != "" {
+				return reconcile.Result{}, nil
+			}
+
+			newCidr := strings.Replace(value, "-", "/", -1)
+			logrus.Infof("No cidr has been set in configmap yet, Setting cidr from namespace label : %v", newCidr)
+
+			config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+				clientcmd.NewDefaultClientConfigLoadingRules(),
+				&clientcmd.ConfigOverrides{CurrentContext: ""}).ClientConfig()
+			if err != nil {
+				logrus.Error(err)
+			}
+
+			// Creates the clientset
+			clientset, err := kubernetes.NewForConfig(config)
+			if err != nil {
+				logrus.Error(err)
+			}
+
+			dataValue := fmt.Sprintf("{ \"production\": { \"createStrategy\": { \"CidrBlock\": \"%v\" } } }", newCidr)
```

Comment:
It's possible that there could be different values inside the `_network` value e.g. https://github.com/integr8ly/cloud-resource-operator/blob/master/deploy/examples/cloud_resources_aws_strategies.yaml#L12-L13 Is there a way that you retrive the data and patch that rather than hardcoding here?

Reply:
let me try and see what I can do